Foundation/FileManager.swift

```diff
@@ -283,13 +283,17 @@ open class FileManager : NSObject {
         result[.size] = NSNumber(value: UInt64(s.st_size))
 
 #if os(OSX) || os(iOS)
-        let ti = (TimeInterval(s.st_mtimespec.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtimespec.tv_nsec))
+        let modificationTime = (TimeInterval(s.st_mtimespec.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtimespec.tv_nsec))
+        let creationTime = (TimeInterval(s.st_ctimespec.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_ctimespec.tv_nsec))
 #elseif os(Android)
-        let ti = (TimeInterval(s.st_mtime) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtime_nsec))
+        let modificationTime = (TimeInterval(s.st_mtime) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtime_nsec))
+        let creationTime = (TimeInterval(s.st_ctime) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_ctime_nsec))
 #else
-        let ti = (TimeInterval(s.st_mtim.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtim.tv_nsec))
+        let modificationTime = (TimeInterval(s.st_mtim.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtim.tv_nsec))
+        let creationTime = (TimeInterval(s.st_ctime.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_ctim.tv_nsec))
```

Comment:
According to http://man7.org/linux/man-pages/man2/lstat.2.html `st_ctime` is not the file's creation time - it is the file's last status change timestamp.

Some Linux filesystems implement `statx.stx_btime` for a file's creation (birth) time but this is not widely supported.

Reply:
Oops.